request/extractor.go

```diff
@@ -79,3 +80,18 @@ func (e *PostExtractionFilter) ExtractToken(req *http.Request) (string, error) {
 		return "", err
 	}
 }
+
+// BearerExtractor extracts a token from the Authorization header.
+// The header is expected to match the format "Bearer XX", where "XX" is the
+// JWT token.
+type BearerExtractor struct{}
```

Comment:
I might have missed it, but why does this need an extra struct? Wouldn't a single function that takes `*http.Request` and returns `string, error` suffice?

Reply:
I assume there is an interface specific to your code base, but I'm not sure this make sense without also defining the interface within this codebase (which I'd rather not do).

Reply:
It follows the extractor interface here https://github.com/golang-jwt/jwt/blob/61ab9b21f8e1ddb45e0f7bf340d0d874408a4cc7/request/extractor.go#L17

Reply:
See [github.com/dgrijalva/jwt-go/v4/request#ParseFromRequest](https://godocs.io/github.com/dgrijalva/jwt-go/v4/request#ParseFromRequest).